### ISSUE TITLE
Provide a public Xur API

### ISSUE THREAD

**Issue by vthornheart-bng** [Bungie-net/api#502]
It has been requested that we revive the old public Xur API!  I think that's a good idea, let's do that when we get the time.

**Comment by joshhunt** [comment]
This would be very helpful!

To display Xur listings on destinysets.com to unauthenticated users, I have to run my own Xur API, which a couple of other people use as well. Problem is maintenance and uptime isn't my strong suit, so it would be great if I didn't have to worry about this any more.

**Comment by janweinen** [comment]
yes please!

**Comment by danilkhromov** [comment]
So there is no way i can get xur inventory without asking users to authenticate?

**Comment by danilkhromov** [comment]
So there is no way i can get xur inventory without asking users to authenticate?

**Comment by mattbaker** [comment]
This would be really nice for bots and the like (really, accessing any vendor), especially given how cumbersome the oauth flow is

**Comment by vthornheart-bng** [comment]
Yes, I agree!  With Triumphs and Forsaken work, this has had to take a back seat for the time being: but it is on our backlog!

**Comment by mattbaker** [comment]
Awesome!! Our future Xur bots thank you :)

**Comment by lowlines** [comment]
This should probably include all vendors, not just Xur. It'd be good if you could specify which vendors you want to return depending on your needs.

**Comment by nine13tech** [comment]
Because of his not specific to a charter nature and the event like status he held, could this not be part of a public vendors endpoint? 
Are their plans for a public vendors endpoint that only shows the static or rotating vendor items that are always for sale or is that a authed endpoint only?

**Comment by ckhicks** [comment]
Does the latest TWAB nullify any of these plans?
```
Xûr
  His will is not his own
    Will no longer display a vendor icon on destination maps
```

**Comment by vthornheart-bng** [comment]
Still to be determined on the endpoint overall - but one way or another, I won't be allowed to return destination information for him.

**Comment by vthornheart-bng** [comment]
Sorry guys - design decisions overrule my preference for more data.

**Comment by justrealmilk** [comment]
But sales data will all be there still?

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by vthornheart-bng** [comment]
Yeah, all of that will still be there in the existing Vendors endpoint.  Still TBD on the idea of having a public Xur endpoint though.

**Comment by ckhicks** [comment]
All good, thanks for keeping us updated. 👍

**Comment by michabbb** [comment]
maybe the [rss feed](https://discussions.ftw.in/c/destiny/find-xur.rss) of findxur.com is a reasonably good solution, until someone provides a real api 🤔

**Comment by michabbb** [comment]
hi all, since finxur.com seems to be down or broken, i am really searching for a reliable source to get this info. is anybody planing something here ? maybe we can create a community build API for this, where the users can report xur´s location with one single click. based on the number of clicks by location, we could take this info as (most) correct info. on the other hand, we could select some public websites and grep the info from there, as soon as it´s available.
is somebody interested in a project like this ? or is there a better idea ?

**Comment by mattybeard** [comment]
> hi all, since finxur.com seems to be down or broken, i am really searching for a reliable source to get this info. is anybody planing something here ? maybe we can create a community build API for this, where the users can report xur´s location with one single click. based on the number of clicks by location, we could take this info as (most) correct info. on the other hand, we could select some public websites and grep the info from there, as soon as it´s available.
> is somebody interested in a project like this ? or is there a better idea ?

https://braytech.org/creators/ and our friend @justrealmilk

**Comment by michabbb** [comment]
@mattybeard nope... braytach changed, the xur api no longer is working, oh boy, this is so stupid of bungie that people have to search for xur´s location every f... week 😠 every stupid friday i need to search google to find his location instead of using the bungie api 😕

**Comment by ArkahnX** [comment]
Lets keep our comments polite and friendly here, Bungie is providing us with a huge service by allowing us to use their APIs, and even so much as officially supporting them this time around, so lets keep things friendly!

**Comment by justrealmilk** [comment]
@mattybeard friend!
@michabbb noticed your issue on my repo and discovered github notifications. 
if it really is so desired, I guess I could apply some priority time to xur and api.braytech.org stuff
I didn't bother redmaking the page these past weeks during my transition to react because xur's inventory is all old boring stuff. I still keep his sales data updated but to save me from manually adjusting his location data after the fact, I've been up at 3am (reset time for Brisbane, Australia) waiting for reddit to find him before changing a PHP variable and running the script that adds the week to my database.

**Comment by michabbb** [comment]
@justrealmilk i guess you don´t have to be awake for that ;) i have a cronjob myself that starts severel times after xurs location should be visible. i searched a few blogs that could be parsed easily (the html dom) - now i have found another one that works, as long they use the same layout. the point is: yes, nobody cares about xurs and his boring old stuff. for me it´s only a part that belongs to the weekly stuff where i notify my community about changes. most don´t care about xur, but some do. it´s annyoing that some things never found their way to the api, like the calus raid counter order, which weapons you can get on the EP this week. the community always has to find workarounds for this stupid things, that are important to many players. 

that´s why i asked here: maybe it´s an idea to create a public api for everybody with infos, bungie is unable to provide, accessable for every developer, managed by the community. 

it´s just an idea.....

**Comment by floatingatoll** [comment]
To clarify something, Bungie isn’t unable — Bungie is _unwilling_.

Thorn indicated that game designers consciously removed Xûr’s map icon in-game and prohibited exposing Xûr’s location in the API.

Please remember to be respectful of Bungie’s choices. Their choices on “how important are API checkpoints to fix” and “whether or not to show Xûr’s location at all” may be unpalatable, but that’s no reason to call their decisions “stupid things”.

**Comment by floatingatoll** [comment]
Vendal’s reply above explicitly says that the game designers have prohibited the official API from publishing Xûr’s location:

https://github.com/Bungie-net/api/issues/502#issuecomment-415820742

**Comment by justrealmilk** [comment]
I said nothing of his location 🙈

**Comment by floatingatoll** [comment]
In two previous replies in this thread you specifically talked about location and being disappointed in Bungie for withholding it. I apologize for confusing your prior comments on this specific issue with your most recent comments on this specific issue, and will mute and exit this thread immediately.

**Comment by vthornheart-bng** [comment]
Ah, you're talking about the fact that we don't have a public Xur endpoint rather than the location info?  In that situation, it's a matter of the time that we have vs. the higher priority work that we have to get to.  In the end, you're talking right now to the only person actively working on the Destiny API.  Whatever I have the ability to do in my workday is what gets done in the API: that's balanced between

* Other external API asks
* Non-Destiny-related Bungie.net work
* Work that has to be done to maintain the API when new game releases break us or change systems in a way that we have to handle
* Internal work (tools, import, our content management system, etc...)

The public Xur endpoint is something I want to provide!  I haven't had time to work on it yet, but it is something I would personally put high on the winter wishlist - so hopefully we can make it happen during the upcoming holiday season!

(But just to note, it won't have location no matter what: indeed I have been asked not to return that information)

**Comment by VeraxonHD** [comment]
> (But just to note, it won't have location no matter what: indeed I have been asked not to return that information)
Confused about this. Surely determining the location of Xur is, like, really easy? As in, just look at where the flashpoint is and go to the same position as he always is on that planet? Why not include that information?

Also any update to this? I don't think @justrealmilk 's braytech.org api has been updated in a while so this would be real useful about now.

**Comment by justrealmilk** [comment]
@VeraxonHD Xur isn’t tied to flashpoint any longer. 

As for my private endpoint, I don’t personally use it at this point and as I have no insights as to its use by others, I find it really difficult to feel invested in maintaining it.

**Comment by VeraxonHD** [comment]
> @VeraxonHD Xur isn’t tied to flashpoint any longer.
Ah, right, of course.
> As for my private endpoint, I don’t personally use it at this point and as I have no insights as to its use by others, I find it really difficult to feel invested in maintaining it.
I used it, for lack of any other alternative, and it was really helpful considering I haven't the time to boot up the game and enter it manually. But that's understantable.

**Comment by vthornheart-bng** [comment]
Aye, I was specifically asked not to return location information in the endpoint unfortunately.  I know that's a bummer for those who use it - sorry that I can't provide it!

Now that it's the week between Christmas and New Year's, I finally have some time to carve out for winter wishlist work.  This one's the next one down in my hit list!  I'll let you know when I've got something ready, and the details of what it is.

**Comment by vthornheart-bng** [comment]
TFS 330631

**Comment by vthornheart-bng** [comment]
Okay, so after a bit of hassle trying to find a more generic way to potentially return more information than just Xur, I found that character-agnostic Vendor information is even hairier now than I realized.  I'm going to return just Xur from this endpoint, who is currently the only vendor that I'm certain doesn't have meaningful character-specific sale items.  Someday I'd like to talk more on the subject, but I fear if I did that I'd go too deep into the secret sauce.  I'd likely be summarily executed.

Anyways, this is ready for the January deployment - but for the forseeable future it will only have Xur in it.  Which I guess is all anyone wanted, but I was personally hoping for more.  Oh well.
